python/huggingfaceserver/huggingfaceserver/__main__.py

```diff
@@ -185,6 +204,7 @@ def load_model():
                 do_lower_case=not kwargs.get("disable_lower_case", False),
                 add_special_tokens=not kwargs.get("disable_special_tokens", False),
                 max_length=kwargs["max_length"],
+                dtype=dtype,
```

Comment:
For encoder models we probably do not want to default to FP16, see
```
2024-05-03 18:13:23.626 1 kserve ERROR [inference_error_handler():104] Exception:
  Traceback (most recent call last):
    File "/huggingfaceserver/huggingfaceserver/encoder_model.py", line 242, in predict
      outputs = self._model(**input_batch).logits
    File "/prod_venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
      return self._call_impl(*args, **kwargs)
    File "/prod_venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
      return forward_call(*args, **kwargs)
    File "/prod_venv/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 1360, in forward
      outputs = self.bert(
    File "/prod_venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
      return self._call_impl(*args, **kwargs)
    File "/prod_venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
      return forward_call(*args, **kwargs)
    File "/prod_venv/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 1006, in forward
      embedding_output = self.embeddings(
    File "/prod_venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
      return self._call_impl(*args, **kwargs)
    File "/prod_venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
      return forward_call(*args, **kwargs)
    File "/prod_venv/lib/python3.10/site-packages/transformers/models/bert/modeling_bert.py", line 239, in forward
      embeddings = self.LayerNorm(embeddings)
    File "/prod_venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1518, in _wrapped_call_impl
      return self._call_impl(*args, **kwargs)
    File "/prod_venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1527, in _call_impl
      return forward_call(*args, **kwargs)
    File "/prod_venv/lib/python3.10/site-packages/torch/nn/modules/normalization.py", line 196, in forward
      return F.layer_norm(
    File "/prod_venv/lib/python3.10/site-packages/torch/nn/functional.py", line 2543, in layer_norm
      return torch.layer_norm(input, normalized_shape, weight, bias, eps, torch.backends.cudnn.enabled)
  RuntimeError: "LayerNormKernelImpl" not implemented for 'Half'
```

Reply:
Actually a lot of CPU-based operations in Pytorch are not implemented to support FP16, so we probably need to check need GPU availability.

Reply:
Thats a fair point. But people would probably want to run models at lower precisions on CPU given that there's not much to lose. We should at least support that right? And for those who want to run, its for themselves to verify compatibility/support?

Reply:
so should we default to float32/float16 based on GPU availability?
something like 
```
default_dtype = 'auto' if torch.cuda.is_available() else 'float32'
dtype = kwargs.get("dtype", default_dtype)
```
Edit: Ok I just checked vLLM CPU support document. For CPU vLLM casts FP16 to BF16... 
https://arc.net/l/quote/nujjbyig
<img width="797" alt="image" src="https://github.com/kserve/kserve/assets/39181234/d4c1ab0c-a678-43fb-8b7a-65d7f8a961ca">